azure-iot-device/azure/iot/device/iothub/sync_clients.py

```diff
@@ -51,8 +51,10 @@ def __init__(self, **kwargs):
         This initializer should not be called directly.
         Instead, use one of the 'create_from_' classmethods to instantiate
 
-        TODO: How to document kwargs?
-        Possible values: iothub_pipeline, edge_pipeline
+        :param iothub_pipeline: The IoTHubPipeline used for the client
+        :type iothub_pipeline: :class:`azure.iot.device.iothub.pipeline.IoTHubPipeline`
+        :param edge_pipeline: The EdgePipeline used for the client
```

Comment:
>edge_pipeline [](start = 15, length = 13)

Not part of this review, but I really dislike this name.  Having the 2 pipeline args makes it look like you chose one pipeline if you're connecting to iothub and a different pipeline if you're connecting to edgehub.

Reply:
I also dislike this name, but have yet to come up with something better. Open to suggestion.

---
In reply to: [334942864](https://github.com/Azure/azure-iot-sdk-python/pull/349#discussion_r334942864) [](ancestors = 334942864)